### PR TITLE
all: formatting code with golangci-lint

### PR DESCRIPTION
**Pull request by armfazh**
enables two linters that automatically format code using 
```
$ make lint-fix
```

Changes:
* dbca74c Linking crypto hashes everywhere is needed.
* 3109fbe Moving file for tests to testdata folder.
* 3aa1ced Formatting code with linter.

### CODE REVIEW

**Review by bwesterb — APPROVED** [review]

File: ecc/bls12381/g1.go [hunk 1]
```diff
@@ -2,7 +2,7 @@ package bls12381
 
 import (
 	"crypto"
-	_ "crypto/sha256" // to link library
```
Comment (Lekensteyn) [inline ecc/bls12381/g1.go:5]: Is this (removed) comment / import still necessary? There are several other users:
```
$ git grep -nEe '_ "crypto/(sha256|sha512)'
blindsign/blindrsa/blindrsa_test.go:8:  _ "crypto/sha256"
ecc/bls12381/g1.go:5:   _ "crypto/sha256" // to link library
expander/expander_test.go:6:    _ "crypto/sha256" // to link libraries
expander/expander_test.go:7:    _ "crypto/sha512" // to link libraries
group/ristretto255.go:5:        _ "crypto/sha512" // to link libraries
hpke/kembase.go:6:      _ "crypto/sha256" // linking sha256 packages.
hpke/kembase.go:7:      _ "crypto/sha512" // linking sha512 packages.
hpke/shortkem.go:6:     _ "crypto/sha256" // linking sha256 packages.
hpke/shortkem.go:7:     _ "crypto/sha512" // linking sha512 packages.
hpke/xkem.go:6: _ "crypto/sha256" // linking sha256 packages.
hpke/xkem.go:7: _ "crypto/sha512" // linking sha512 packages.
```

File: blindsign/blindrsa/blindrsa_test.go [hunk 2]
```diff
@@ -362,7 +362,7 @@ func verifyTestVector(t *testing.T, vector testVector) {
 }
 
 func TestVectors(t *testing.T) {
-	data, err := ioutil.ReadFile("test_vectors.json")
+	data, err := ioutil.ReadFile("testdata/test_vectors.json")
```
Comment (Lekensteyn) [inline blindsign/blindrsa/blindrsa_test.go:365]: Note for reviewer: this file indeed got moved.

**Review by Lekensteyn — APPROVED** [review]
Some open questions about the `crypto/sha256` and `crypto/sha512` imports where an import comment was dropped. Changes look reasonable in any case, but you might want to update the commit message with the kind of changes (moved json file to testdata directory, formatting changes using `<whatever command you used>`) such that it can be reproduced.

File: ecc/bls12381/g1.go (hunk 1 above)
Comment (armfazh) [reply ecc/bls12381/g1.go:5]: I reviewed all places where linking libraries is needed. thanks for pointing this

File: blindsign/blindrsa/blindrsa_test.go (hunk 2 above)
Comment (armfazh) [reply blindsign/blindrsa/blindrsa_test.go:364]: I created a single commit for this change

**Pull request merged**